### PR TITLE
[DONT MERGE] Test making Slice artificially address-only

### PR DESCRIPTION
**Pull request by jckarter**
Experimenting with potential resilience mitigations to allow for future optimization of in-place modifications through slices.

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please benchmark

File: stdlib/public/core/Slice.swift [hunk 1]
```diff
@@ -135,6 +135,9 @@ public struct Slice<Base: Collection> {
   public var base: Base {
     return _base
   }
+
+  /// Test cost of making Slice artificially address-only
```
Comment (slavapestov) [inline stdlib/public/core/Slice.swift]: It would be better to add an underscored attribute for this.

Comment (jckarter) [reply stdlib/public/core/Slice.swift]: Sure, I'm just trying an experiment to see if it's even viable. Adding an `Any` field also approximates making the value witness operations opaque.

Comment (slavapestov) [reply stdlib/public/core/Slice.swift]: Well, a worst case test would be to just remove the @_fixed_layout attribute wouldn’t it?

**Pull request closed without merging**